fft/src/polynomial.rs

```diff
@@ -1,76 +1,86 @@
 use lambdaworks_math::{
-    field::{element::FieldElement, traits::IsFFTField},
-    helpers,
+    field::{
+        element::FieldElement,
+        traits::{IsFFTField, RootsConfig},
+    },
     polynomial::Polynomial,
 };
 
-use crate::{
-    errors::FFTError,
-    ops::{fft_with_blowup, inverse_fft},
-};
+#[cfg(feature = "metal")]
+use lambdaworks_gpu::metal::fft::polynomial::evaluate_fft_metal;
+
+use crate::{errors::FFTError, roots_of_unity::get_twiddles};
 
 pub trait FFTPoly<F: IsFFTField> {
-    fn evaluate_fft(&self) -> Result<Vec<FieldElement<F>>, FFTError>;
+    fn evaluate_fft(
+        &self,
+        blowup_factor: usize,
+        domain_size: Option<usize>,
+    ) -> Result<Vec<FieldElement<F>>, FFTError>;
```

Comment:
Should `blowup_factor` be an `Option<usize>` which defaults to 1 if `None`? just a nit